book/src/exec.md

```diff
@@ -0,0 +1,174 @@
+This note will only be spent if the user's intent (intentVP) is satisfied.
+2. Once all of the parties involved in a state transition spent their intent token notes (meaning that the intents of all of them are satisfied), the transaction is balanced and can be finalized.
+
+![img.png](img/exec_intent_notes.png)
+
+
+If we look at the intent token notes as messages passing from one user (application) to another, the output [1] intent note can signal a message being sent, and spending of that note will signal that the message has been received. See more about the idea of message passing [here](message_passing.md).
+
+## Step 2: Gossip
+Once the intentVP is specified and the initial partial transaction is created, the user sends everything to the intent gossip network. Gossip nodes send those intents to each other, spreading them around. Solvers listen to the intent gossip network and receive the intents.
+
+## Step 3: Solve
+
+Solvers receive intents from the intent gossip network and match them together in order to create transactions. 
+
+That implies that users need to give some information to the solvers that is sufficient to create partial transactions and transactions (to create proofs). In order to match two intents, the solver needs to know:
+- intentVPs of both matching parties
+- note commitments and nullifiers from prior partial transactions (to be chained)
+- proofs from prior partial transactions (to be chained)
+- for the notes spent/output in the new partial transaction: nullifier key, commitments to the sender/receiver VPs, note fields
+
+We are considering the model where a solver makes one step at a time and sends the step result back to the gossip network, where the partial solution meets the next solver in the chain. In practice, the solver can continue solving instead of sending the result back to the gossip network if they can make the next step.
+
+### Create partial transactions
+
+When a solver has intents that can be [partially] matched together, they [partially] match the intents by creating new partial transactions. Solvers cannot modify the old data, only create new (e.g. new partial transactions).
+
+#### Prove
+Solvers are responsible for creation of all proofs (`Action`, `VP`, etc) required for their partial solutions. For that reason, solvers:
+- **have the authority** to spend and create the notes they receive and produce the proofs required to perform the action,
+- **know** the content of intentVPs of the users (necessary to be able to satisfy them),
+- **don't know** the identities of the users.
+
+When solvers receive partial solutions from the gossip network, they must check all of the proofs attached to them.
+
+#### Local `cm` trees
+
+To store note commitments of the created notes, a local commitment tree `CMtree` is created. After the transaction is finalized, the tree will be published on the blockchain along with the transaction.
```

Comment:
We cannot just consider the final transaction and forget about the partial ones (and so we don't consider the local `CMTree`)?

Reply:
To create proofs for intermediate ptx (when you spend a note that was created in the previous ptx, it doesn't have `cm` published in the global `CMtree`) we need to have local`CMtree`

Reply:
Mhhh I see, I forgot we need to spend the ptx :-D